lib/iris/coords.py

```diff
@@ -446,8 +447,12 @@ def is_full_slice(s):
             points = self._points
             bounds = self._bounds
         else:
-            points = self.points
-            bounds = self.bounds
+            points = self._points
+            if isinstance(points, iris.aux_factory.LazyArray):
+                points = points.view()
```

Comment:
@rhattersley This forces the deferred coordinate data to actually load and cache, right?

If so, perhaps a comment here might be useful ...